### PR TITLE
Feature/incident-email

### LINKED ISSUE #370: Send email to client and incident submitter after incident submit

### PR DESCRIPTION
**Pull request by pnobbe**
### Merge when:

* [x] The merge build succeeded
* [ ] The PR has been reviewed and is approved
* [ ] All packages are upgraded to the lastest version (only PR into master)

**What issue does this PR address?**

Fixes #370 

**Does this PR introduce a breaking change?**

No

**Additional information**:

None

### CODE REVIEW

File: src/FunderMaps.BatchNode/Jobs/Notification/NotificationTask.cs [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+            if (envelope == null)
+            {
+                throw new ProtocolException();
+            }
+
+            await NotifyAsync(context, envelope);
+        }
+
+        /// <summary>
+        ///     Method to check if a task can be handeld by this job.
```
Comment (pnobbe) [inline src/FunderMaps.BatchNode/Jobs/Notification/NotificationTask.cs:35]:
```suggestion
        ///     Method to check if a task can be handled by this job.
```

File: src/FunderMaps.BatchNode/Jobs/Notification/NotificationTask.cs [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        /// </summary>
+        /// <param name="name">The task name.</param>
+        /// <param name="valdirectoryNameue">The task payload.</param>
```
Comment (pnobbe) [inline src/FunderMaps.BatchNode/Jobs/Notification/NotificationTask.cs:38]:
```suggestion
        /// <param name="value">The task payload.</param>
```

File: src/FunderMaps.BatchNode/Startup.cs [hunk 3]
```diff
@@ -38,6 +38,7 @@ public void ConfigureServices(IServiceCollection services)
 
             // Add batch jobs to the DI container.
             services.AddBatchJob<Jobs.BundleBuilder.BundleJob>();
+            services.AddBatchJob<Jobs.Notification.EmailHandler>();
```
Comment (pnobbe) [inline src/FunderMaps.BatchNode/Startup.cs:41]:
```suggestion
            services.AddBatchJob<Jobs.Notification.EmailJob>();
```

File: src/FunderMaps.Infrastructure/BatchClient/BatchClient.cs [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+using FunderMaps.Core.Interfaces;
+using System.Threading.Tasks;
+using System.Threading;
+
+namespace FunderMaps.Infrastructure.BatchClient
+{
+    /// <summary>
+    ///     Client connector to the batch node.
+    /// </summary>
+    internal class BatchClient : Batch.BatchClient, IBatchService // TODO: Inherit from AppServiceBase
```
Comment (pnobbe) [inline src/FunderMaps.Infrastructure/BatchClient/BatchClient.cs:12]: Do not inherit from BatchClient

```suggestion
    internal class BatchClient : IBatchService // TODO: Inherit from AppServiceBase
```

File: src/FunderMaps.Infrastructure/BatchClient/ChannelFactory.cs [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+        /// <summary>
+        ///     Get a remote channel to the GRPC service.
+        /// </summary>
+        public ChannelBase RemoteChannel
+        {
+            get
+            {
+                // The port number(5001) must match the port of the gRPC server.
+                return GrpcChannel.ForAddress("https://localhost:5001", new GrpcChannelOptions
```
Comment (pnobbe) [inline src/FunderMaps.Infrastructure/BatchClient/ChannelFactory.cs:30]: Address should be retrieved from config

File: src/FunderMaps.Portal/Controllers/IncidentPortalController.cs [hunk 6]
```diff
@@ -103,8 +107,12 @@ public async Task<IActionResult> CreateIncidentAsync([FromBody] IncidentDto inpu
             }
 
             await _incidentRepository.AddAsync(incident);
-
-            // TODO: Dispatch notification
+            await _notifyService.DispatchNotify(new Envelope
+            {
+                Recipients = new List<string> { "info@fundermaps.com", "info@laixer.com" },
```
Comment (pnobbe) [inline src/FunderMaps.Portal/Controllers/IncidentPortalController.cs:112]: List should come from a config file

File: src/FunderMaps.Core/Notification/INotifyService.cs [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+    /// <summary>
+    ///     Notify service.
+    /// </summary>
+    public interface INotifyService
+    {
+        /// <summary>
+        ///     Notify by means of contacting.
+        /// </summary>
+        /// <param name="envelope">Envelope containing the notification.</param>
+        Task DispatchNotify(Envelope envelope);
```
Comment (pnobbe) [inline src/FunderMaps.Core/Notification/INotifyService.cs:14]:
```suggestion
        Task DispatchNotifyAsync(Envelope envelope);
```

File: src/FunderMaps.BatchNode/Jobs/Notification/NotificationTask.cs [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+        /// <returns><c>True</c> if method handles task, false otherwise.</returns>
+        public override bool CanHandle(string name, object value)
+        {
+            if (!(name.ToLowerInvariant() == TaskName && value is string))
+            {
+                return false;
+            }
```
[the remark this answers is not in the document]
Comment (yorickdewid) [reply src/FunderMaps.BatchNode/Jobs/Notification/NotificationTask.cs:45]: Nope, we can also do partial string match

File: src/FunderMaps.Core/Notification/INotifyService.cs (hunk 7 above)
Comment (yorickdewid) [reply src/FunderMaps.Core/Notification/INotifyService.cs:14]: Done

**Review by yorickdewid — APPROVED** [review]

**Pull request merged**